src/qibo/backends/qulacs.py

```diff
@@ -0,0 +1,81 @@
+import re
+
+import numpy as np
+
+from qibo import __version__
+from qibo.backends import NumpyBackend
+from qibo.result import CircuitResult, QuantumState
+
+
+class QulacsBackend(NumpyBackend):
+
+    def __init__(self):
+        super().__init__()
+        import qulacs  # pylint: disable=import-error
+        from qulacs import (  # pylint: disable=no-name-in-module, import-error
+            QuantumCircuitSimulator,
+            converter,
+        )
```

Comment:
Why not top-level?

You could cut the `import` chain by scoping the `import` of this backend inside `qibo/backends/__init__.py`, but it'd make sense that if you `import` the Qulacs backend, you will import Qulacs as well :P

Reply:
I did not remember why I had not imported it top level, so, I tried again, and the problem is that the import in `/qibo/backends/__init__.py` will break if `qulacs` is not installed, [https://github.com/qiboteam/qibo/actions/runs/9663894252/job/26657235334?pr=1272](https://github.com/qiboteam/qibo/actions/runs/9663894252/job/26657235334?pr=1272)

better to have it inside then, I would say.

Reply:
Definitely, I agree.

But we can make it non-top here:
https://github.com/qiboteam/qibo/blob/94d5c89cd8bf13e969f9abf432854a8d8b3e18c9/src/qibo/backends/__init__.py#L43-L44
and break the chain.

Reply:
ah ok I see, thus you are proposing of adding the imports to `__init__.py`?

``` python
elif backend == "qulacs":
    import qulacs  # pylint: disable=import-error
    from qulacs import (  # pylint: disable=no-name-in-module, import-error
        QuantumCircuitSimulator,
        converter,
     )
     return QulacsBackend()
```

It looks just a little bit weird to me that you import things here that you use in `qulacs.py`, but I am fine with this in case.

Reply:
ah yeah, of course, that was so simple indeed 🤦‍♂️